tests/e2e/unbonding.go

```diff
@@ -139,68 +140,114 @@ func (s *CCVTestSuite) TestUndelegationNoValsetChange() {
-	checkStakingUnbondingOps(s, 1, true, true)
-	// - check that CCV unbonding op was created
-	checkCCVUnbondingOp(s, s.providerCtx(), s.consumerChain.ChainID, valsetUpdateID, true)
-
-	// call NextBlock on the provider (which increments the height)
-	s.providerChain.NextBlock()
-
-	// check that the VSC packet is stored in state as pending
-	pendingVSCs, _ := providerKeeper.GetPendingVSCs(s.providerCtx(), s.consumerChain.ChainID)
-	s.Require().True(len(pendingVSCs) == 1, "no pending VSC packet found")
-
-	// delegate again to create another VSC packet
-	delegate(s, delAddr, bondAmt)
-
-	// call NextBlock on the provider (which increments the height)
-	s.providerChain.NextBlock()
-
-	// check that the VSC packet is stored in state as pending
-	pendingVSCs, _ = providerKeeper.GetPendingVSCs(s.providerCtx(), s.consumerChain.ChainID)
-	s.Require().True(len(pendingVSCs) == 2, "only one pending VSC packet found")
-
-	// increment time so that the unbonding period ends on the provider
-	incrementTimeByUnbondingPeriod(s, Provider)
-	// - check that the unbonding op is still there and onHold is true
-	checkStakingUnbondingOps(s, 1, true, true)
-	// - check that unbonding has not yet completed, i.e., the initBalance
-	// is still lower by the bond amount, because it has been taken out of
-	// the delegator's account
-	s.Require().True(getBalance(s, s.providerCtx(), delAddr).Equal(initBalance.Sub(bondAmt).Sub(bondAmt)))
-
-	// complete CCV channel setup
-	s.SetupCCVChannel()
-	s.SetupTransferChannel()
-
-	// relay VSC packets from provider to consumer
-	relayAllCommittedPackets(s, s.providerChain, s.path, ccv.ProviderPortID, s.path.EndpointB.ChannelID, 2)
-
-	// increment time so that the unbonding period ends on the consumer
-	incrementTimeByUnbondingPeriod(s, Consumer)
-
-	// relay VSCMatured packets from consumer to provider
-	relayAllCommittedPackets(s, s.consumerChain, s.path, ccv.ConsumerPortID, s.path.EndpointA.ChannelID, 2)
-
-	// check that the unbonding operation completed
-	// - check that ccv unbonding op has been deleted
-	checkCCVUnbondingOp(s, s.providerCtx(), s.consumerChain.ChainID, valsetUpdateID, false)
-	// - check that staking unbonding op has been deleted
-	checkStakingUnbondingOps(s, valsetUpdateID, false, false)
-	// - check that one quarter the delegated coins have been returned
-	s.Require().True(getBalance(s, s.providerCtx(), delAddr).Equal(initBalance.Sub(bondAmt).Sub(bondAmt.Quo(sdk.NewInt(2)))))
+	testCases := []struct {
+		name                       string
+		updateInitTimeoutTimestamp func(*providerkeeper.Keeper, time.Duration)
+		removed                    bool
+	}{
+		{
+			"channel handshake completes before unbonding period", func(pk *providerkeeper.Keeper, pUnbondingPeriod time.Duration) {
+				// change the init timeout timestamp for this consumer chain
+				// to make sure the chain is not removed before the unbonding period elapses
+				ts := s.providerCtx().BlockTime().Add(pUnbondingPeriod + 24*time.Hour)
```

Comment:
Fix this by making it so channel handshake completes after unbonding period but before init timeout period